source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1049,11 +1049,11 @@ void ShaderGraph::optimize(GenContext& context)
         {
             // Check if we have a constant conditional expression
             ShaderInput* intest = node->getInput("intest");
-            if (!intest->getConnection() || intest->getConnection()->getNode()->hasClassification(ShaderNode::Classification::CONSTANT))
+            if (!intest->getConnection())
```

Comment:
Don't bypass the conditional if the connected node is a constant node, since the constant node can now be connected further upstream and we can't guarantee that it's constant.

So only bypass now if there is an unconnected conditional input.